### PR TITLE
nlohmann-json: 3.6.1 (new formula)

### PR DESCRIPTION
**Pull request by JarryShaw**
`nlohmann/json` is a (seems) wildly used JSON
library for C++. The formula is directly migrated
from this [Tap](https://github.com/nlohmann/homebrew-json).
See: https://github.com/nlohmann/json

### CODE REVIEW

**Comment by SMillerDev** [comment]
Optional dependencies are not acceptable in homebrew-core. I'd suggest just using cmake support.

**Comment by SMillerDev** [comment]
You can drop the caveat now that there's no options anymore. And I don't think all the explanation in the test is required. Other maintainers might disagree though.

**Comment by JarryShaw** [comment]
Ah… my bad 🤦‍♂️
Imma keep the tests for now and wait for u guys to review :)

**Review by SMillerDev — APPROVED** [review]

File: Formula/nlohmann-json.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        // add an object inside the object
+        j["answer"]["everything"] = 42;
+
+        // add an array that is stored as std::vector (using an initializer list)
+        j["list"] = { 1, 0, 2 };
+
+        // add another object (using an initializer list of pairs)
+        j["object"] = { {"currency", "USD"}, {"value", 42.99} };
+
+        // instead, you could also write (which looks very similar to the JSON above)
```
Comment (fxcoudert) [inline Formula/nlohmann-json.rb]: Let's simplify the test. Remove comments, let's keep only a few objects in the json (say `pi`, `name`, `list` and `object`). Serialise the resulting json and match the output of that program (using `shell_output` and `assert_match`), so we exercise I/O as well.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Review by JarryShaw — COMMENTED** [review]
Simplified tests based on review per @fxcoudert .

**Comment by fxcoudert** [comment]
@JarryShaw I don't think you've pushed your changes to the test, they do not appear

**Comment by JarryShaw** [comment]
> @JarryShaw I don't think you've pushed your changes to the test, they do not appear

Sorry again 🤦‍♂️ not familiar with this GitHub feature

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @JarryShaw this is looking good!

**Pull request closed without merging**